### PR TITLE
Fix mistral workflow state determination

### PR DESCRIPTION
**Pull request by m4dcoder**
Fix use case where workflow is canceled but tasks is in idle state. Clean up state determination for additional use cases.

### CODE REVIEW

**Comment by DoriftoShoes** [comment]
LGTM

File: contrib/runners/mistral_v2/query/mistral_v2.py [hunk 1]
```diff
@@ -133,18 +137,22 @@ def _determine_execution_status(self, execution_id, wf_state, tasks):
         is_action_canceled = action_service.is_action_canceled_or_canceling(execution_id)
 
         # Identify the list of tasks that are not in completed states.
```
Comment (bigmstone) [inline contrib/runners/mistral_v2/query/mistral_v2.py]: Can you update or remove this comment?

File: contrib/runners/mistral_v2/tests/unit/test_mistral_querier_v2.py [hunk 2]
```diff
@@ -188,10 +193,18 @@ def test_determine_status_wf_canceled_tasks_running(self):
...
+        self.assertEqual(action_constants.LIVEACTION_STATUS_CANCELED, status)
+
+    @mock.patch.object(
+        action_service, 'is_action_canceled_or_canceling',
+        mock.MagicMock(return_value=True))
     def test_determine_status_wf_canceled_exec_running_tasks_completed(self):
         wf_id = uuid.uuid4().hex
         status = self.querier._determine_execution_status(wf_id, 'RUNNING', MOCK_WF_TASKS_SUCCEEDED)
-        self.assertEqual(action_constants.LIVEACTION_STATUS_RUNNING, status)
+        self.assertEqual(action_constants.LIVEACTION_STATUS_CANCELING, status)
```
Comment (bigmstone) [inline contrib/runners/mistral_v2/tests/unit/test_mistral_querier_v2.py]: :heart:

**Comment by m4dcoder** [comment]
Comment updated. Thanks!

**Comment by lakshmi-kannan** [comment]
👍

**Pull request merged**